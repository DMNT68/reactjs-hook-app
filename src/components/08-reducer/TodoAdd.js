import React from 'react';
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChage, reset] = useForm({
    description: '',
  });

  const handleSubmit = (e) => {
    e.preventDefault();

    if (description.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      desc: description,
      done: false,
    };
    handleAddTodo(newTodo);
    reset();
  };

  return (
    <div>
      <h4>Agregar</h4>
      <hr />

      <form onSubmit={handleSubmit}>
        <input value={description} onChange={handleInputChage} className="form-control" type="text" name="description" placeholder="Aprender..." autoComplete="off" />
        <div className="d-grid gap-2">
          <button type="submit" className="btn btn-outline-primary mt-4">
            Agregar
          </button>
        </div>
      </form>
    </div>
  );
};
