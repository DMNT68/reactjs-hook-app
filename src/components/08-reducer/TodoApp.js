import React, { useReducer } from 'react';
import { todoReducer } from './todoReducer';

import './styles.css';

const initialState =[{
    id: new Date().getTime(),
    desc: 'Aprender react',
    done: false
}];

export const TodoApp = () => {

    const [todos] = useReducer(todoReducer, initialState)

    return (
        <div>
            <h1>TODO APP</h1>
            <ul>
                <li>Hola</li>
                <li>Mundo</li>
                <li>Hola de nuevo</li>
            </ul>
        </div>
    )
}
