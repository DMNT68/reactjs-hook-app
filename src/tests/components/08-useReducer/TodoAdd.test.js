import React from 'react';
import { shallow } from 'enzyme';
import { TodoAdd } from '../../../components/08-reducer/TodoAdd';

describe('Pruebas en <TodoAdd />', () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);
  test('debe de mostrase correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('NO debe de llamar el handleAddTodo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test('debe de llamar la funcion handleAddTodo', () => {
    const value = 'Aprender React';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      },
    });
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
        id: expect.any(Number),
        desc: value,
        done: false
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
