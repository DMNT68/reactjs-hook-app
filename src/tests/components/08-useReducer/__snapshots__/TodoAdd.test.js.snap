// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> debe de mostrase correctamente 1`] = `
<div>
  <h4>
    Agregar
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2"
    >
      <button
        className="btn btn-outline-primary mt-4"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</div>
`;
